js/piwik.js

```diff
@@ -7281,6 +7289,33 @@ if (typeof window.Piwik !== 'object') {
                 this.requireConsent();
             };
 
+            /**
+             * Detects whether a user has opted out.
+             *
+             * @returns {boolean}
+             */
+            this.isUserOptedOut = isUserOptedOut;
+
+            /**
+             * Opts a user out from tracking. If called, no tracker requests will be sent to Matomo for the current
+             * user.
+             *
+             * Note: using the opt out strategy conflicts with the consent features, since here we are assuming
+             * consent. It also follows that depending on where you/your users are located, using opt-out instead
+             * requiring consent may affect how well you comply with relevant privacy laws (ie, GDPR).
+             */
+            this.optUserOut = function () {
+                setCookie('piwik_ignore', '*', null, configCookiePath, configCookieDomain, configCookieIsSecure);
```

Comment:
Just wondering: should it also prevent setting any other Matomo cookies similar to the consent feature? And when the consent feature deletes all cookies, would it also remove a set ignore cookie?

Reply:
👍 for preventing cookies getting set.

I was thinking originally that consent would be mutually exclusive w/ opt-out, but now it makes more sense to me to have both. Ie, require consent & provide option to just say no. But making them work together isn't necessarily straightforward...

Here are the scenarios I can think of:
1. user consents, then later opts out: in this case we delete the consent cookie & set the opt out cookie
2. user opts out w/o consenting: in this case, we set the piwik opt out cookie, and clear the consent queue
3. user opts out (before or after consenting), then consents later: is this a sort of "opt-in"? or do they need to also give their consent? i'm thinking in this case, the opt-in cookie should be removed.

Which also makes me think if a user opted out, the consent queue should never be modified.

So that would mean the following changes:
- when setting opt-out, remove consent cookie & clear consent queue
- when opt-out cookie exists, do not add to consent queue
- when consenting, remove opt-out cookie
- (and make sure when opted-out no other cookies set)